src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,41 +1,122 @@
-const fetchFileExtension = /(.*)\.[^\.]*/;
-const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
-const fetchDoubleWhitespacesRegExp = /  */gm;
+import {fetchAllMatchesForAGroup} from "../regExpUtils";
+import builtInPredicates from "./builtInPredicates";
 
 export function removeFileExtension(path: string) {
+	const fetchFileExtension = /(.*)\.[^\.]*/;
+
 	const fetch: string[] | null = fetchFileExtension.exec(path);
 	if (fetch === null) { return ""; }
 	return fetch[1];
```

Comment:
What happens if fetch is an array of size 1? Then the only entry it has is entry 0, isn't it? Returning entry 1 shouldn't work in that case.

Reply:
Good question. I just checked the behavior of that.

It works out though. The variable `fetch` will always be either `null` or `string[1]` will be defined.
Index 0 refers to the whole match and index 1 refers to the first group. So one limit case would be as example:
`.pl` which will just evaluate to an empty string then. Group 1 will just be empty.
In case of no match it will return null.

Reply:
So change the regex to `/(.+)\.[^\.]*/` then.